### DISCUSSION TITLE
limit the number of revision

### DISCUSSION THREAD

**Discussion by xinyou-app** [area17/twill#917] [Q&A] [answered]
to avoid the database grown faster, Isn't able to set the limited revision for each item?

**Comment by pboivin** [comment, marked as answer]
Hi @xinyou-app,

I don't know if there's a better way to do it, but here's a quick example to limit the number of revisions for a given module. You can add the following to your module repository :

```
    const MAX_REVISIONS = 4;

    public function afterSave($object, $fields)
    {
        parent::afterSave($object, $fields);

        $object->refresh()->revisions->slice(self::MAX_REVISIONS)->each->delete();
    }
```

This was not tested extensively but I hope it can be helpful!

**Reply by xinyou-app** [reply]
Thanks, this is helpful.

**Comment by ifox** [comment]
Great suggestion @pboi20! That operation can also be pushed to a queue to avoid any slowdown on save.

Another option would be to schedule an artisan command.

**Comment by domihagen** [comment]
It would be great to have this functionality in the core of Twill and be able to control it via the config.
```php
'max_revisions' => 10, // default: false
```
